### DISCUSSION TITLE
"new row violates row-level security policy" error for every `insert` call

### DISCUSSION THREAD

**Discussion by TheOnlyTails** [supabase-community/supabase-kt#31] [Q&A] [answered]
I have this `insert` call which runs every time a user sends a message in a discord channel: 
```kt
client.postgrest["scores"].insert(
    User(it.author.name, discordId = it.author.idLong),
    upsert = true,
).decodeAs<User>()
```
but every time I send to trigger it, I get this error message: 
```
io.github.jan.supabase.exceptions.RestException: Unknown error: {"code":"42501","details":null,"hint":null,"message":"new row violates row-level security policy for table \"scores\""} (HTTP status 401) 
```

**Comment by jan-tennert** [comment, marked as answer]
It might be that you only have access to inserting in the table but not selecting, so try to set the returning parameter to minimal (which won't return the changed row when calling decodeAs) or change the rls policy. But it seems like you want to receive the updated data so you probably have to look in your policies

**Reply by TheOnlyTails** [reply]
I get the data so I can increment the score later on, but I don't have any RLS policies, so IDK why this is happening

**Reply by jan-tennert** [reply]
So RLS is generally disabled for "scores"?

**Reply by TheOnlyTails** [reply]
It's enabled, there just aren't any policies.
![image](https://user-images.githubusercontent.com/65342367/204056193-cedfff98-0716-47e7-b207-9e84443ea0fd.png)

**Reply by jan-tennert** [reply]
If you have RLS enabled, and have no policies, supabase will not allow any requests to the table as there are no policies. You'd have to disable RLS if you don't have policies. Also there's no point in enabling it if you don't need it

**Reply by TheOnlyTails** [reply]
Alright, that works!
